app/lib/synthetic_branding_group_populator.rb

```diff
@@ -0,0 +1,107 @@
+  BASE_BRANDING_GROUPS_CONFIG_PATH = Rails.root.join('db', 'seed', 'synthetic_branding_groups')
+
+  # populate all branding groups using synthetic studies
+  #
+  # @param user (User)                 => User account to own the branding groups
+  # @param overwrite_groups (Boolean)  => T/F for destroying/recreating any existing groups
+  # @param overwrite_studies (Boolean) => T/F for destroying/recreating any existing studies
+  def self.populate_all(user: User.first, overwrite_groups: true, overwrite_studies: false)
+    # populate all studies first to ensure they exist
+    SyntheticStudyPopulator.populate_all(overwrite: overwrite_studies)
+    synthetic_studies_ids = self.collect_synthetic_studies.pluck(:id)
+    # divide into two groups, one for human data, another for mouse
+    human_studies = Study.where(name: /human/i, :id.in => synthetic_studies_ids)
+    mouse_studies = Study.where(name: /(mouse|murine)/i, :id.in => synthetic_studies_ids)
+    if human_studies.empty? || mouse_studies.empty?
+      raise ArgumentError.new("Required synthetic studies not present, please manually run SyntheticStudyPopulator.populate_all")
+    end
+    group_dirs = Dir.glob(BASE_BRANDING_GROUPS_CONFIG_PATH.join('*')).select {|f| File.directory? f}
+    group_dirs.each do |group_dir|
+      # detect if this is the human or mouse group
+      group_type = group_dir.split('/').last.split('_').first
+      study_list = group_type == 'human' ? human_studies : mouse_studies
+      self.populate(group_dir, user: user, study_list: study_list, overwrite: overwrite_groups)
+    end
+  end
+
+  # remove all synthetic branding groups, leaving studies in place
+  def self.remove_all
+    config_dirs = Dir.glob(BASE_BRANDING_GROUPS_CONFIG_PATH.join('*')).select {|f| File.directory? f}
+    config_dirs.each do |config_dir|
+      branding_group_config = self.parse_configuration_file(config_dir)
+      group_name = branding_group_config['branding_group']['name']
+      group = BrandingGroup.find_by(name: group_name)
+      if group.present?
+        puts "Removing branding group #{group_name}"
+        group.destroy
+      end
+    end
+  end
+
+  # create a single branding group and associate designated studies
+  #
+  # @param config_dir (Pathname, String)  => Directory pathname containing configuration JSON for branding group
+  # @param user (User)                    => User account to own branding group
+  # @param study_list (Mongoid::Criteria) => List of studies to associate with new branding group
+  # @param overwrite (Boolean)            => T/F for destroying/recreating an existing group
+  def self.populate(config_dir, user: User.first, study_list:, overwrite: true)
+    branding_group_config = self.parse_configuration_file(config_dir)
+
+    puts("Populating synthetic branding group from #{config_dir}")
+    branding_group = create_branding_group(branding_group_config, user, overwrite: overwrite)
+    # assign new branding group to all studies
```

Comment:
This update_all shouldn't happen if overwrite is false, and the study already existed, right?

Reply:
This was required because of the corner case where you had _some_ synthetic studies, but not all.  I needed to be sure that you would assign everything.  Even if you aren't overwriting anything, there was still the possibility that a study was created.  If that logic changes, then yes - this would be true here.